pkg/controller/controller.go

```diff
@@ -46,7 +48,19 @@ func AddToManager(ctx context.Context,
 			csr.ControllerName,
 			func() error {
 				return csr.Add(ctx, manager, clientHolder,
-					[]func(ctx context.Context, csr *certificatesv1.CertificateSigningRequest) (bool, error){})
+					[]func(ctx context.Context, csr *certificatesv1.CertificateSigningRequest) (bool, error){
+						// Case 1: If flightctl is enabled and a csr is from a flightctl device, approve it.
+						func(ctx context.Context, csr *certificatesv1.CertificateSigningRequest) (bool, error) {
+							isFlightCtlEnabled, err := flightctler.IsFlightCtlEnabled(ctx)
+							if err != nil {
+								return false, err
+							}
+							if isFlightCtlEnabled {
+								return flightctler.IsManagedClusterAFlightctlDevice(ctx, csr.Spec.Username)
```

Comment:
what if there is an error here? how can the csr reonciler to be triggered again?

Reply:
We refactor the csr controller before, if it returns err, if will requeue: 

https://github.com/stolostron/managedcluster-import-controller/blob/f65821a693aabfb51b10a2cbf28a5e8db407e3e6/pkg/controller/csr/csr_controller.go#L104C1-L114C3